cognite/client/_api/functions.py

```diff
@@ -0,0 +1,1136 @@
+    assert not (
+        has_function_id and has_function_external_id
+    ), "Only function_id or function_external_id allowed when listing schedules."
+
+
+def extract_requirements_from_file(file_name: str) -> List[str]:
+    """Extracts a list of library requirements from a file. Comments, lines starting with '#', are ignored.
+
+    Args:
+        file_name (str): name of the file to parse
+
+    Returns:
+        (list[str]): returns a list of library records
+    """
+    requirements: List[str] = []
+    with open(file_name, "r+") as f:
+        for line in f:
+            line = line.strip()
+            if UNCOMMENTED_LINE_REG.match(line):
+                requirements.append(line)
+    return requirements
+
+
+def extract_requirements_from_doc_string(docstr: str) -> Optional[List[str]]:
+    """Extracts a list of library requirements defined between [requirements] and [/requirements] in a functions docstring.
+
+    Args:
+        docstr (str): the docstring to extract requirements from
+
+    Returns:
+        (list[str] | None): returns a list of library records if requirements are defined in the docstring, else None
+    """
+    substr_start, substr_end = None, None
+
+    # Get index values for the start and end of the requirements list
+    for match in REQUIREMENTS_REG.finditer(docstr):
+        val = match.group()
+        if val == "[requirements]":
+            substr_start = match.end()
+        elif val == "[/requirements]":
+            substr_end = match.start()
+
+    if substr_start and substr_end:
+        # Return a list of requirement entries
+        return docstr[substr_start:substr_end].splitlines()[1:]
+    return None
+
+
+def validate_requirements(requirements: List[str]) -> str:
+    """Validates the requirement specifications
+
+    Args:
+        requirements (list[str]): list of requirement specifications
+
+    Raises:
+        ValueError: if validation of requirements fails
+
+    Returns:
+        str: output path of the requirements file
+    """
+    parsed_reqs: List[str] = []
+    for req in requirements:
+        try:
+            parsed = install_req_from_line(req)
+        except Exception as e:
+            raise ValueError(str(e))
+
+        parsed_reqs.append(str(parsed).strip())
+
+    tmp = NamedTemporaryFile()
+
+    # Write requirements to temporary file
+    with open(tmp.name, "w+") as f:
+        f.write("\n".join(parsed_reqs))
+        f.close()
+
+    return tmp.name
+
+
+def get_requirements_handle(fn: Callable) -> Optional[str]:
+    """Read requirements from a function docstring, and validate them
+
+    Args:
+        fn (Callable): the function to read requirements from
+
+    Returns:
+        str: output path of the requirements file, or None if no requirements are specified
+    """
+    docstr = getdoc(fn)
+
+    if docstr:
+        reqs = extract_requirements_from_doc_string(docstr)
+        if reqs:
+            return validate_requirements(reqs)
+
+    return None
```

Comment:
Is making these functions public deliberate?

Reply:
@HMEiding these were the functions you were talking about right?
 
 Yea these should be private.

Reply:
Yup, should be private.